docs/docs/userguide/configuration/CONFIGURATION.md

```diff
@@ -0,0 +1,185 @@
+   
+The [mandatory configuration](#mandatory-configuration) items are those you should define once before the first installation. Mandatory values cannot be changed during the entire environment lifecycle. 
+
+The [optional configuration](#optional-configuration) items are not required for installation by default. Optional values may change at any point in the environment lifecycle.
+Terraform will retain the latest state of the environment and keep track of any configuration changes made later.
+
+The following is an example of a valid configuration file:
+
+``` terraform
+# Mandatory items
+environment_name = "my-bamboo-env"
+region           = "us-east-2"
+
+# Optional items
+resource_tags = {
+  Terraform    = "true",
+  Organization = "atlassian",
+  product      = "bamboo" ,
+}
+
+instance_types   = ["m5.xlarge"]
+desired_capacity = 2
+domain           = "mydomain.com"
+```
+
+## Common Configuration
+
+Environmental properties common to all deployments.
+
+### Environment Name
+
+`environment_name` provides your environment a unique name within a single cloud provider account. This value cannot be altered after the configuration has been applied. The value will be used to form the name of some resources including `VPC` and `Kubernetes cluster`.
+
+```terraform
+environment_name = "<your-environment-name>" # e.g. "my-terraform-env"
+```
+
+!!! info "Format" 
+    
+    Environment names should start with a letter and can contain letters, numbers, and dashes (`-`). The maximum value length is 24 characters.
+
+
+### Region
+
+`region` defines the cloud provider region that the environment will be deployed to.
+
+```terraform
+region = "<REGION>"  # e.g. "ap-northeast-2"
+```
+
+!!! info "Format"
+
+    The value must be a valid [AWS region](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Concepts.RegionsAndAvailabilityZones.html){.external}.
+
+### Products
+
+The `products` list can be configured with one or many products. This will result in these products being deployed to the same K8s cluster. For example, if a Jira and Confluence deployment is required this property can be configured as follows:
+
+```terraform
+products = ["jira", "confluence"]
+```
+
+!!! info "Product specific infrastructure"
+
+    All of the appropriate infrastructure for the product selection will be stood up by Terraform.
+
+### Domain
+
+We recommend using a domain name to access the application via `HTTPS`. You will be required to secure a domain name and supply the configuration to the config file.
+
+When the domain is provided, Terraform will create a [Route53](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/Welcome.html) hosted zone based on the `environment` name.
+
+```terraform
+domain="<DOMAIN NAME>" # e.g. "mydomain.com"
+```
+
+A fully qualified domain name uses the following format: `<product>.<environment-name>.<domain-name>`. For example `bamboo.staging.mydomain.com`.
+
+!!! tip "Ingress controller"
```

Comment:
I have removed a note that the domain cannot be removed after it is provisioned - I have tested that yesterday and it was possible.

Just drive by change ;) 👦 scout rule